tests/sha1_test.c

```diff
@@ -0,0 +1,256 @@
+    (void)ctx;
+
+    struct aws_byte_cursor input =
+        aws_byte_cursor_from_c_str("abcdbcdecdefdefgefghfghighijhijkijkljklmklmnlmnomnopnopq");
+    uint8_t expected[] = {
+        0x84, 0x98, 0x3e, 0x44, 0x1c, 0x3b, 0xd2, 0x6e, 0xba, 0xae,
+        0x4a, 0xa1, 0xf9, 0x51, 0x29, 0xe5, 0xe5, 0x46, 0x70, 0xf1,
+    };
+    struct aws_byte_cursor expected_buf = aws_byte_cursor_from_array(expected, sizeof(expected));
+
+    return s_verify_hash_test_case(allocator, &input, &expected_buf, aws_sha1_new);
+}
+
+AWS_TEST_CASE(sha1_nist_test_case_3, s_sha1_nist_test_case_3_fn)
+
+static int s_sha1_nist_test_case_4_fn(struct aws_allocator *allocator, void *ctx) {
+    (void)ctx;
+
+    struct aws_byte_cursor input = aws_byte_cursor_from_c_str("abcdefghbcdefghicdefghijdefghijkefghijklfghij"
+                                                              "klmghijklmnhijklmnoijklmnopjklmnopqklm"
+                                                              "nopqrlmnopqrsmnopqrstnopqrstu");
+    uint8_t expected[] = {
+        0xa4, 0x9b, 0x24, 0x46, 0xa0, 0x2c, 0x64, 0x5b, 0xf4, 0x19,
+        0xf9, 0x95, 0xb6, 0x70, 0x91, 0x25, 0x3a, 0x04, 0xa2, 0x59,
+    };
+    struct aws_byte_cursor expected_buf = aws_byte_cursor_from_array(expected, sizeof(expected));
+
+    return s_verify_hash_test_case(allocator, &input, &expected_buf, aws_sha1_new);
+}
+
+AWS_TEST_CASE(sha1_nist_test_case_4, s_sha1_nist_test_case_4_fn)
+
+static int s_sha1_nist_test_case_5_fn(struct aws_allocator *allocator, void *ctx) {
+    (void)ctx;
+
+    aws_cal_library_init(allocator);
+
+    struct aws_hash *hash = aws_sha1_new(allocator);
+    ASSERT_NOT_NULL(hash);
+    struct aws_byte_cursor input = aws_byte_cursor_from_c_str("a");
+
+    for (size_t i = 0; i < 1000000; ++i) {
+        ASSERT_SUCCESS(aws_hash_update(hash, &input));
+    }
+
+    uint8_t output[AWS_SHA1_LEN] = {0};
+    struct aws_byte_buf output_buf = aws_byte_buf_from_array(output, sizeof(output));
+    output_buf.len = 0;
+    ASSERT_SUCCESS(aws_hash_finalize(hash, &output_buf, 0));
+
+    uint8_t expected[] = {
+        0x34, 0xaa, 0x97, 0x3c, 0xd4, 0xc4, 0xda, 0xa4, 0xf6, 0x1e,
+        0xeb, 0x2b, 0xdb, 0xad, 0x27, 0x31, 0x65, 0x34, 0x01, 0x6f,
+    };
+    struct aws_byte_cursor expected_buf = aws_byte_cursor_from_array(expected, sizeof(expected));
+    ASSERT_BIN_ARRAYS_EQUALS(expected_buf.ptr, expected_buf.len, output_buf.buffer, output_buf.len);
+
+    aws_hash_destroy(hash);
+
+    aws_cal_library_clean_up();
+
+    return AWS_OP_SUCCESS;
+}
+
+AWS_TEST_CASE(sha1_nist_test_case_5, s_sha1_nist_test_case_5_fn)
+
+static int s_sha1_nist_test_case_5_truncated_fn(struct aws_allocator *allocator, void *ctx) {
+    (void)ctx;
+
+    aws_cal_library_init(allocator);
+
+    struct aws_hash *hash = aws_sha1_new(allocator);
+    ASSERT_NOT_NULL(hash);
+    struct aws_byte_cursor input = aws_byte_cursor_from_c_str("a");
+
+    for (size_t i = 0; i < 1000000; ++i) {
```

Comment:
Is 1000000 significant? Seems like it should be a local constant with maybe a comment.